homeassistant/components/media_player/kodi.py

```diff
@@ -184,10 +184,17 @@ def _check_deprecated_turn_off(hass, turn_off_action):
         if properties is not None:
             unique_id = properties.get('uuid', None)
 
+    # Skip unresolvable hosts. This has been previously "handled"
+    # by an unhanled exception thrown by socket.gethostbyname()
+    try:
+        socket.getaddrinfo(host, port)
+    except socket.gaierror as exc:
+        _LOGGER.error("Can't resolve hostname '%s', skipping: %r", host, exc)
+        return
+
     # Only add a device once, so discovered devices do not override manual
     # config.
-    ip_addr = socket.gethostbyname(host)
-    if ip_addr in hass.data[DATA_KODI]:
+    if host in hass.data[DATA_KODI]:
```

Comment:
This breaks things. The reason for resolving the host is so that we have a single way of identifying machines. If a manual config specifies a host, and then we discover an IP, we don't want to end up with 2 kodi's set up.

Reply:
FWIW it looks like the upstream uuid change has been merged for Kodi v18 https://github.com/xbmc/xbmc/pull/14089

We might be able to skip the IP check if we have a uuid to use, to allow IPv6 for Kodi versions 18 and above.